andela/vof-deployment-scripts
# #154790526 Replace Target pools with GCLB cookies

#### What does this PR do?
This PR removes a feature target pool that had been added previously to the repo and replaces it with another feature GCLB cookie.
#### Description of Task to be completed?
Currently on vof there is an issue on the csrf token on staging. Previous workaround was to implement target pools  that would allow session stickiness and prevent a 422 error when creating, updating or deleting objects on the vof application. This was occurring because of the round robin algorithm of the google cloud platform load balancer. This would lead to POST, PUT and DELETE requests not sending a csrf token that the instance would recognize after it has been previously generated by another instance. Target pools showed results. However due to reasons not yet determined, it did not work as expected. This led to a fix of implementing session affinity using session cookies. 
#### How should this be manually tested?
Request from several clients should be sent to the application using different browsers and accounts. Logging information should be checked on the instances to determine where a particular client's request was processed. With a number of requests sent repeatedly from a particular client, results should show that one client's request fall on the same instance throughout. This occurrence will prove the existence of load balancer stickiness.

Second test would involve going through routes that involve a POST, PUT or DELETE request. Ensure that all requests are processed or if not processed that the error does not claim that no access has been given to perform the action.
#### Any background context you want to provide?
The implemented solution on GCLB Cookie can be found on https://cloud.google.com/compute/docs/load-balancing/http/backend-service .
The removed solution on the target pool can be found on https://cloud.google.com/compute/docs/load-balancing/network/target-pools

It should be noted however that affinity can be lost and such scenarios should be mitigated.
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/154790526
